ee/clickhouse/queries/column_optimizer.py

```diff
@@ -104,7 +105,10 @@ def properties_used_in_filter(self) -> Counter[PropertyIdentifier]:
             #
             # See ee/clickhouse/queries/trends/util.py#process_math
             if entity.math == "unique_group":
-                counter[(f"$group_{entity.math_group_type_index}", "event", None)] += 1
+                if self.is_actor_query:
```

Comment:
Not sure this if-else is correct.

It's saying if it's an actor query, property `$group_{N}` for group type math_group_type_index is used, which does not seem correct.